### PR TITLE
Call entry functions using a loop

### PR DESCRIPTION
**Pull request by poechsel**
Before this PR `caml_program` was a sequence of function call and variable incrementation. It turns out that when linking large binaries a sizable amount of the linking time was actually spend generating this function (between the register allocation and assembling it).
This PR changes `caml_program` as follow:
- Store the list of entry symbol addresses in `.data`
- Loop other the symbol and call them one after another

### CODE REVIEW

**Review by gretay-js — APPROVED** [review]
looks good to me, a couple of minor comments.

**Comment by chambart** [comment]
If this is a problem for you, I would suggest to look into potentially removing those calls in the first place:
There are many, if not a majority of OCaml modules with empty initialization function: It would be simple to detect that this is only a function returning unit. In that case, we could just avoid calling it.

File: backend/cmm_helpers.ml [hunk 1]
```diff
@@ -3601,27 +3601,90 @@ let entry_point namelist =
...
+        Csymbol_address
+          (global_symbol (make_symbol ~compilation_unit:name "entry")))
+      namelist
+  in
+  let data = Cdefine_symbol table_symbol :: data in
+  let raise_num = Lambda.next_raise_count () in
+  let id_prev = VP.create (V.create_local "*id_prev*") in
+  let id = VP.create (Ident.create_local "i") in
+  let high = cconst_int (List.length namelist - 1) in
   let body =
```
[the remark this answers is not in the document]
Comment (poechsel) [reply backend/cmm_helpers.ml:3646]: Done

File: backend/cmm_helpers.ml [hunk 2]
```diff
@@ -3601,27 +3601,90 @@ let entry_point namelist =
...
-      fun_codegen_options = [Reduce_code_size];
-      fun_dbg;
-      fun_poll = Default_poll
-    }
+  [ Cdata data;
+    Cfunction
+      { fun_name;
+        fun_args = [];
+        fun_body = Csequence (body, cconst_int 1);
+        fun_codegen_options = [Reduce_code_size];
```
[the remark this answers is not in the document]
Comment (poechsel) [reply backend/cmm_helpers.ml]: Good idea

File: backend/cmm_helpers.ml [hunk 3]
```diff
@@ -3582,7 +3582,21 @@ let make_symbol ?compilation_unit name =
...
+ *     caml_globals_entry_functions[i]();
+ *     caml_globals_inited += 1;
+ *     int idprev = i;
+ *     id += 1;
+ *     if (idprev == len_caml_globals_entry_functions) goto out;
+ *   }
+ *   out:
+ *   return 1;
+ * }
+ *)
```
Comment (gretay-js) [inline backend/cmm_helpers.ml:3599]: `s/idprev/id_prev/`

**Review by lthls — APPROVED** [review]
Looks fine to me

**Pull request merged**